core/src/main/java/org/infinispan/statetransfer/CommitManager.java

```diff
@@ -68,11 +69,41 @@ public final void stopTrack(Flag track) {
          }
          tracker.clear();
       } else {
-         for (Iterator<Map.Entry<Object, DiscardPolicy>> iterator = tracker.entrySet().iterator();
-              iterator.hasNext(); ) {
-            if (iterator.next().getValue().update(trackStateTransfer, trackXSiteStateTransfer)) {
-               iterator.remove();
-            }
+         for (Map.Entry<Integer, Map<Object, DiscardPolicy>> segment : tracker.entrySet()) {
+            clearSegmentKeys(segment.getKey(),
+                  policy -> policy.getValue().update(trackStateTransfer, trackXSiteStateTransfer));
+         }
+      }
+   }
```

Comment:
simpler code :) 
avoids doing: iterate -> Map.get -> Map.remove
```suggestion
       tracker.values().removeIf(entries -> {
            entries.values().removeIf(discardPolicy -> discardPolicy.update(trackStateTransfer, trackXSiteStateTransfer));
            return entries.isEmpty();
         });
```